tests/test_extensions.py

```diff
@@ -310,6 +310,17 @@ def testRE(self):
             self.assertEqual(RE.match(test).groups(), expected)
 
 
+class TestTOCAndNl2br(unittest.TestCase):
+
+    def setUp(self):
+        self.md = markdown.Markdown(extensions=['toc', 'nl2br'])
+
+    def testNotToc(self):
+        """ Test TOC with a single newline is not populated. """
+        text = '[TOC]\ntext'
+        self.assertEqual(self.md.convert(text), '<p>[TOC]<br />\ntext</p>')
```

Comment:
In addition to the points made by @facelessuser, the new test case should be using the newer [markdown.test_tools.TestCase][1] and be located in [tests/test_syntax/extensions/test_toc.py][2]

[1]: https://python-markdown.github.io/test_tools/#markdowntest_toolstestcase
[2]: https://github.com/Python-Markdown/markdown/blob/master/tests/test_syntax/extensions/test_toc.py